lib/ex_sdp.ex

```diff
@@ -68,5 +62,55 @@ defmodule ExSDP do
 
   defdelegate parse(text), to: Parser
   defdelegate parse!(text), to: Parser
-  defdelegate serialize(session), to: Serializer
+
+  @spec new(version :: non_neg_integer(), origin :: binary(), session_name :: binary()) :: t()
+  def new(version, origin, session_name) do
+    %__MODULE__{
+      version: version,
+      origin: origin,
+      session_name: session_name
+    }
+  end
+
+  @spec set_timing(sdp :: t(), timing :: Timing.t()) :: t()
+  def set_timing(sdp, timing) do
+    Bunch.Struct.put_in(sdp, :timing, timing)
+  end
+
+  @spec add_media(sdp :: t(), media :: Media.t() | [Media.t()]) :: t()
+  def add_media(sdp, media) do
+    media = sdp.media ++ Bunch.listify(media)
+    Bunch.Struct.put_in(sdp, :media, media)
```

Comment:
```suggestion
    Map.update!(sdp, :media, & &1 ++ Bunch.listify(media))
```

and you can add `use Bunch.Access` to SDP anyway